### PR TITLE
Simplify logic in boundedMemoryQueue, use channels len/cap

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: exporter/exporterhelper/internal/bounded_memory_queue_test.go [hunk 1]
```diff
@@ -246,12 +246,3 @@ func (s *consumerState) assertConsumed(expected map[string]bool) {
 	}
 	assert.Equal(s.t, expected, s.snapshot())
 }
-
-func TestZeroSize(t *testing.T) {
```
Comment (bogdandrutu) [inline exporter/exporterhelper/internal/bounded_memory_queue_test.go:250]: This is covered by not allowing users to configure 0 size, see QueueConfig.Validate()

File: exporter/exporterhelper/internal/bounded_memory_queue_test.go [hunk 2]
```diff
@@ -247,11 +247,24 @@ func (s *consumerState) assertConsumed(expected map[string]bool) {
 	assert.Equal(s.t, expected, s.snapshot())
 }
 
-func TestZeroSize(t *testing.T) {
+func TestZeroSizeWithConsumers(t *testing.T) {
```
Comment (bogdandrutu) [inline exporter/exporterhelper/internal/bounded_memory_queue_test.go:250]: There is a small behavior change here if capacity/size is 0 and numConsumers >0. But this is not a valid case in QueueConfig, see Validate so we should not worry.

**Review by yurishkuro — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Submitted https://github.com/open-telemetry/opentelemetry-collector/pull/8832 to make the test pass

File: exporter/exporterhelper/internal/bounded_memory_queue.go [hunk 3]
```diff
@@ -91,11 +75,11 @@ func (q *boundedMemoryQueue) Stop() {
 
 // Size returns the current size of the queue
 func (q *boundedMemoryQueue) Size() int {
-	return int(q.size.Load())
+	return len(q.items)
```
Comment (atoulme) [inline exporter/exporterhelper/internal/bounded_memory_queue.go:78]: If we are waiting on the call a back on a queue item, is it still in the queue?

**Comment by dmitryax** [comment]
@bogdandrutu please rebase

**Pull request merged**